src/AppInstallerCLIE2ETests/TestCommon.cs

```diff
@@ -125,13 +126,16 @@ public static RunCommandResult RunAICLICommandViaInvokeCommandInDesktopPackage(s
             }
 
             string workDirectory = GetRandomTestDir();
+            string tempBatchFile = Path.Combine(workDirectory, "Batch.cmd");
             string exitCodeFile = Path.Combine(workDirectory, "ExitCode.txt");
             string stdOutFile = Path.Combine(workDirectory, "StdOut.txt");
             string stdErrFile = Path.Combine(workDirectory, "StdErr.txt");
 
-            cmdCommandPiped += $"{AICLIPath} {command} {parameters} > {stdOutFile} 2> {stdErrFile} & call echo %^ERRORLEVEL% > {exitCodeFile}";

```

Comment:
>^ [](start = 112, length = 1)

I remember somehow I added this ^ on purpose. I think it's related to not expanding the variable during parsing time, but only expand it during execution time. I don't exactly remember how it works now.

But if adding \n makes it not expanding during parsing time, that might be cleaner

Reply:
Yes, I believe that in interactive mode, this allows the value to not be evaluated until later.  In a batch script, you can simply place it on the next line to prevent that.

---
In reply to: [519081614](https://github.com/microsoft/winget-cli/pull/633#discussion_r519081614) [](ancestors = 519081614)